src/script.cpp

```diff
@@ -1531,11 +1553,24 @@ bool is_script_code_sig_type(const operation_stack& ops)
         return false;
     }
     const operation_stack& code_ops = script_code.operations();
-    // Minimum size is 4
+    const payment_type redeem_type = script_code.type();
     // M [SIG]... N checkmultisig
+    // or one of the other standard transaction types
     return code_ops.size() >= 4 &&
```

Comment:
Thanks for implementing this. I may not have time to review it myself in the next few days, but I'd like to get a review from some of the other maintainers, because it's in script.